### ISSUE TITLE
Killed?

### ISSUE THREAD

**Issue by KNLargadas** [ssec/polar2grid#291]
Hello again. I have an Ubuntu 20.04 running on VirtualBox and I got this error, says "Line 45: 2826 Killed". This is the whole thing:

billyswift@billyswift: /example_for_modis_1$ polar2grid.sh crefl gtiff --true-color --compress=NONE -g wgs84_modis --fornav-D 10 -f /home/billyswift/example_for_modis_1
INFO     : Initializing reader...
INFO     : Could not find any existing crefl output will use MODIS SDRs to create some
INFO     : Creating navigation product 'longitude_1000m'
INFO     : Creating navigation product 'latitude_1000m'
INFO     : Creating data product 'solar_zenith_angle'
INFO     : Initializing remapping...
INFO     : Initializing backend...
INFO     : Initializing compositor objects...
INFO     : Extracting swaths from data files available...
WARNING  : Missing proper data files to create product: viirs_crefl03
WARNING  : Missing proper data files to create product: viirs_crefl04
WARNING  : Missing proper data files to create product: viirs_crefl01
WARNING  : Missing proper data files to create product: viirs_crefl08
INFO     : Creating navigation product 'longitude_250m'
INFO     : Interpolating to higher resolution: Longitude
INFO     : Creating navigation product 'latitude_250m'
INFO     : Creating navigation product 'longitude_500m'
INFO     : Interpolating to higher resolution: Longitude
INFO     : Creating navigation product 'latitude_500m'
INFO     : Creating data product 'modis_crefl03_500m'
INFO     : Creating data product 'modis_crefl01_500m'
INFO     : Creating data product 'modis_crefl01_250m'
INFO     : Creating data product 'modis_crefl04_500m'
INFO     : Remapping to grid wgs84_modis
INFO     : Grid information:
	grid_name: wgs84_modis
	proj4_definition: +proj=latlong +datum=WGS84 +ellps=WGS84 +no_defs
	cell_width: 0.00225
	cell_height: -0.00225
	width: 12811
	height: 9441
	origin_x: 123.43620300292969
	origin_y: 40.5906982421875
INFO     : Running gridded scene through 'true_color' compositor
INFO     : Saving true color image to filename 'grid_wgs84_modis_true_color.dat'
INFO     : Creating output from data mapped to grid wgs84_modis
INFO     : Creating geotiff 'terra_modis_modis_crefl03_500m_20180610_014000_wgs84_modis.tif'
INFO     : Creating geotiff 'terra_modis_modis_crefl01_500m_20180610_014000_wgs84_modis.tif'
INFO     : Creating geotiff 'terra_modis_modis_crefl04_500m_20180610_014000_wgs84_modis.tif'
INFO     : Creating geotiff 'terra_modis_modis_crefl01_250m_20180610_014000_wgs84_modis.tif'
/home/billyswift/polar2grid_v_2_3/bin/polar2grid.sh: line 45:  2826 Killed                  python3 -m polar2grid.glue_legacy $@ -vv
billyswift@billyswift: /example_for_modis_1$ python3 -m polar2grid.glue_legacy $@ -vv
/usr/bin/python3: Error while finding module specification for 'polar2grid.glue_legacy' (ModuleNotFoundError: No module named 'polar2grid')

Please help me out with this one.

**Comment by djhoese** [comment]
Hi there. The "Killed" message is usually your system telling you it had to kill a process due to it using too much memory. How much memory did you give your virtual machine?

One thing that seems odd to me is based on log output it looks like it finished writing all but the true_color geotiff. I would have guessed it would have failed much sooner than that (in resampling). Then again, the true_color composite is the largest one so it isn't completely surprising.

Anyway, if you try increasing the amount of memory you give to the virtual machine then you should have better luck.

**Comment by KNLargadas** [comment]
It's around 5191 MB, near the red zone. Is it safe to push it higher?

**Comment by djhoese** [comment]
It depends how much you have on your system and if you are sharing this system with anyone else. How much memory does the host system have? If this is your personal machine and you aren't doing anything else with it then temporarily bumping it to a higher value shouldn't be a problem. If you plan on using this VM for continuous processing then you'll want to give it enough memory to have it finish its just while also leaving the host machine enough memory to be used. I believe we recommend 8 or maybe 16GB of RAM for Polar2Grid.

**Comment by KNLargadas** [comment]
Hi again. I downgraded to Ubuntu 16.04 and created a new one. And I encountered this error:

billyswift@billyswift:~/example_for_modis_1$ polar2grid.sh crefl gtiff --true-color --compress=NONE -g wgs84_modis --fornav-D 10 -f /home/billyswift/example_for_modis_1
INFO     : Initializing reader...
INFO     : Could not find any existing crefl output will use MODIS SDRs to create some
INFO     : Creating navigation product 'longitude_1000m'
INFO     : Creating navigation product 'latitude_1000m'
INFO     : Creating data product 'solar_zenith_angle'
INFO     : Initializing remapping...
INFO     : Initializing backend...
INFO     : Initializing compositor objects...
INFO     : Extracting swaths from data files available...
WARNING  : Missing proper data files to create product: viirs_crefl08
WARNING  : Missing proper data files to create product: viirs_crefl03
WARNING  : Missing proper data files to create product: viirs_crefl01
WARNING  : Missing proper data files to create product: viirs_crefl04
INFO     : Creating navigation product 'longitude_500m'
INFO     : Interpolating to higher resolution: Longitude
INFO     : Creating navigation product 'latitude_500m'
INFO     : Creating navigation product 'longitude_250m'
INFO     : Interpolating to higher resolution: Longitude
INFO     : Creating navigation product 'latitude_250m'
INFO     : Creating data product 'modis_crefl01_500m'
INFO     : Creating data product 'modis_crefl03_500m'
INFO     : Creating data product 'modis_crefl01_250m'
INFO     : Creating data product 'modis_crefl04_500m'
INFO     : Remapping to grid wgs84_modis
INFO     : Grid information:
	grid_name: wgs84_modis
	proj4_definition: +proj=latlong +datum=WGS84 +ellps=WGS84 +no_defs
	cell_width: 0.00225
	cell_height: -0.00225
	width: 12811
	height: 9441
	origin_x: 123.43620300292969
	origin_y: 40.5906982421875
INFO     : Running gridded scene through 'true_color' compositor
INFO     : Saving true color image to filename 'grid_wgs84_modis_true_color.dat'
INFO     : Creating output from data mapped to grid wgs84_modis
INFO     : Creating geotiff 'terra_modis_modis_crefl01_500m_20180610_014000_wgs84_modis.tif'
INFO     : Creating geotiff 'terra_modis_modis_crefl03_500m_20180610_014000_wgs84_modis.tif'
INFO     : Creating geotiff 'terra_modis_modis_crefl04_500m_20180610_014000_wgs84_modis.tif'
INFO     : Creating geotiff 'terra_modis_modis_crefl01_250m_20180610_014000_wgs84_modis.tif'
ERROR    : Unexpected error. Enable debug messages (-vvv) or see log file for details.

Here's the log. This has linux-swap (which is not present in the former error) so I don't know what happened. 
[crefl2gtiff_20180610_014000.log](https://github.com/ssec/polar2grid/files/5141870/crefl2gtiff_20180610_014000.log)
